test/e2e_clutz_dts_test.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// tslint:disable:no-unused-expression mocha .to.be.empty getters.
+
+import {expect} from 'chai';
+import * as ts from 'typescript';
+
+import * as tsickle from '../src/tsickle';
+
+import {compileWithTransfromer, dtsCompilerOptions} from './test_support';
+
+describe('clutz dts', () => {
+  it('does nothing if there are no exports', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `const x = 1;`);
+
+    // Run tsickle+TSC to convert inputs to Closure JS files.
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    expect(dts).not.to.contain('namespace ಠ_ಠ.clutz');
+  });
+
+  it('reexports things in the clutz namespace', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `
+      export const x = 1;
+      export function incr(n) { return x + 1; }`);
+
+    // Run tsickle+TSC to convert inputs to Closure JS files.
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    expect(dts).to.contain('namespace ಠ_ಠ.clutz');
+    expect(dts).to.contain('const module$contents$input_x = x;');
+    expect(dts).to.contain('const module$contents$input_incr = incr;');
+  });
+
+  it('exports the type and value of a class', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `
+      export class foo {}`);
+
+    // Run tsickle+TSC to convert inputs to Closure JS files.
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    expect(dts).to.contain('type module$contents$input_foo = foo;');
+    expect(dts).to.contain('const module$contents$input_foo: typeof foo;');
+  });
+
+  it('handles multisegment paths', () => {
+    const sources = new Map<string, string>();
+    sources.set('path/to/file/input.ts', `
+      export const x = 1;`);
+
+    // Run tsickle+TSC to convert inputs to Closure JS files.
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('path/to/file/input.d.ts')!;
+
+    expect(dts).to.contain('const module$contents$path$to$file$input_x = x;');
+  });
+});
```

Comment:
can we have a test that the output is syntactically valid .d.ts. I am afraid we will miss a closing brace somewhere.

Reply:
Done.